packages/sitecore-jss-react/src/components/PlaceholderCommon.tsx

```diff
@@ -305,10 +305,12 @@ export class PlaceholderCommon<T extends PlaceholderProps> extends React.Compone
       .filter((element) => element); // remove nulls
 
     if (this.props.sitecoreContext?.editMode === EditMode.Metadata) {
+      // default value of uid for root placeholder when uid is not present.
+      const default_UID = '00000000-0000-0000-0000-000000000000';
       return [
         <PlaceholderMetadata
           key={(this.props.rendering as ComponentRendering).uid}
-          uid={(this.props.rendering as ComponentRendering).uid}
+          uid={(this.props.rendering as ComponentRendering).uid || default_UID}
```

Comment:
Let's define instead DEFAULT_UID constant in _PlaceholderMetadata_ component

Reply:
I think what Illia meant here was to define a "const DEFAULT_UID" at the top of the component, that can be changed if default_uid ever needs a change

Reply:
I think having the logic defined as `uid={(this.props.rendering as ComponentRendering).uid || DEFAULT_UID` is more descriptive of the fact that there is a default/fallback ID and there's a possibility of retrieved uid being absent or empty.

Reply:
@art-alexeyenko This logic is related only to _PlaceholderMetadata_ component, ideally it should be incapsulated in specific component, _PlaceholderCommon_ shouldn't be aware of it